test/minikube/minikube_long_restore_test.go

```diff
@@ -891,3 +891,220 @@ func TestKubernetesRestoreDatabaseSeparateJournal(t *testing.T) {
+		}
+
+		if journalSnapshotName != "" {
+			values["database.sm.hotCopy.journalPath.enabled"] = "true"
+			values["database.sm.hotCopy.journalPath.persistence.dataSourceRef.kind"] = "VolumeSnapshot"
+			values["database.sm.hotCopy.journalPath.persistence.dataSourceRef.name"] = journalSnapshotName
+			values["database.sm.hotCopy.journalPath.persistence.dataSourceRef.apiGroup"] = "snapshot.storage.k8s.io"
+			values["database.sm.hotCopy.journalPath.persistence.storageClass"] = testlib.SNAPSHOTABLE_STORAGE_CLASS
+		}
+
+		var databaseChartName string
+
+		if shouldStart {
+			databaseChartName = testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+				SetValues: values,
+			})
+
+			output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", admin0, "-c", "admin", "-c", "admin", "--", "bash", "-c",
+				fmt.Sprintf("echo \"SELECT id FROM testtbl;\" | nuosql %s --user dba --password secret", dbName))
+
+			require.NoError(t, err, output)
+
+			require.True(t, strings.Contains(output, "123"))
+		} else {
+			databaseChartName = testlib.StartDatabaseNoWait(t, namespaceName, admin0, &helm.Options{
+				SetValues: values,
+			})
+
+			smPod := fmt.Sprintf("sm-%s-nuodb-cluster0-%s-hotcopy-0", databaseChartName, dbName)
+			var pod *corev1.Pod
+			testlib.Await(t, func() bool {
+				var err error
+				pod, err = k8s.GetPodE(t, kubectlOptions, smPod)
+				return err == nil && len(pod.Status.ContainerStatuses) > 0 &&
+					pod.Status.ContainerStatuses[0].State.Terminated != nil &&
+					pod.Status.ContainerStatuses[0].State.Terminated.Reason == "Error"
+			}, 60*time.Second)
+			retVal = k8s.GetPodLogs(t, kubectlOptions, pod, "engine")
+		}
+
+		helm.DeleteE(t, &options, databaseChartName, true)
+		return retVal
+	}
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+
+	sourceDb := "src-noj"
+	sourceDatabaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.persistence.storageClass":     testlib.SNAPSHOTABLE_STORAGE_CLASS,
+			"database.name":                         sourceDb,
+		},
+	})
+
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", admin0, "-c", "admin", "--", "bash", "-c",
+		fmt.Sprintf("echo \"CREATE TABLE testtbl (id INT); INSERT INTO testtbl (id) values (123);\" | nuosql %s --user dba --password secret", sourceDb))
+
+	require.NoError(t, err, output)
+
+	defer testlib.Teardown(testlib.TEARDOWN_SNAPSHOT)
+
+	smPod := fmt.Sprintf("sm-%s-nuodb-cluster0-%s-hotcopy-0", sourceDatabaseChartName, sourceDb)
+	achiveVolumeName := "archive-volume-" + smPod
+
+	noJournalNoBidSnapshotName := "noj-nobid-snapshot"
```

Comment:
Can you add comments in here to say what each of these snapshots is testing?

Reply:
Will do.